hack/jenkins/test-flake-chart/compute_flake_rate.go

```diff
@@ -0,0 +1,227 @@
+/*
+Copyright 2021 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"bufio"
+	"flag"
+	"fmt"
+	"io"
+	"os"
+	"runtime/debug"
+	"sort"
+	"strings"
+	"time"
+)
+
+var (
+	dataCsv   = flag.String("data-csv", "", "Source data to compute flake rates on")
+	dateRange = flag.Uint("date-range", 5, "Number of test dates to consider when computing flake rate")
+)
+
+func main() {
+	flag.Parse()
+
+	file, err := os.Open(*dataCsv)
+	if err != nil {
+		exit("Unable to read data CSV", err)
+	}
+
+	testEntries := ReadData(file)
+	splitEntries := SplitData(testEntries)
+	filteredEntries := FilterRecentEntries(splitEntries, *dateRange)
+	flakeRates := ComputeFlakeRates(filteredEntries)
+	fmt.Println("Environment,Test,Flake Rate")
+	for environment, environmentSplit := range flakeRates {
+		for test, flakeRate := range environmentSplit {
+			fmt.Printf("%s,%s,%.2f\n", environment, test, flakeRate*100)
+		}
+	}
+}
+
+type TestEntry struct {
+	name        string
+	environment string
+	date        time.Time
+	status      string
+}
+
+// A map with keys of (environment, test_name) to values of slcies of TestEntry.
+type SplitEntryMap map[string]map[string][]TestEntry
```

Comment:
private or the name of public should be added to the comment

Reply:
Done.